indexer/indexer_tests/feature_metadata_test.cpp

```diff
@@ -0,0 +1,102 @@
+#include "testing/testing.hpp"
+
+#include "indexer/feature_meta.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/writer.hpp"
+
+#include "std/map.hpp"
+#include "std/target_os.hpp"
+
+using feature::Metadata;
+
+UNIT_TEST(Feature_Metadata_GetSet)
+{
+  Metadata m;
+  Metadata::EType const type = Metadata::FMD_ELE;
+  // Absent types should return empty values.
+  TEST_EQUAL(m.Get(type), "", ());
+  m.Set(type, "12345");
+  TEST_EQUAL(m.Get(type), "12345", ());
+  TEST_EQUAL(m.Size(), 1, ());
+  // Same types should replace old metadata values.
+  m.Set(type, "5678");
+  TEST_EQUAL(m.Get(type), "5678", ());
+  // Empty values should drop fields.
+  m.Set(type, "");
+  TEST_EQUAL(m.Get(type), "", ());
+  TEST_EQUAL(m.Size(), 0, ());
+  TEST(m.Empty(), ());
+}
+
+static map<Metadata::EType, string> const kPairs = { {Metadata::FMD_ELE, "12345"},
```

Comment:
We use kNotation for any type of constant in many places. I don't see a better name for it here.
Will move it up though.